kgraphql/src/main/kotlin/com/apurebase/kgraphql/schema/structure/SchemaCompilation.kt

```diff
@@ -409,11 +403,9 @@ class SchemaCompilation(
             throw SchemaException("Invalid union type members")
         }
 
-        val __typenameField = handleOperation(
-            PropertyDef.Function<Nothing, String?>("__typename", FunctionWrapper.on({ value: Any ->
-                schemaProxy.typeByKClass(value.javaClass.kotlin)?.name
-            }, true))
-        )
+        val __typenameField = typenameField(FunctionWrapper.on({ value: Any ->
+            queryTypeProxies[value.javaClass.kotlin]?.name ?: error("No query type proxy found for $value")
```

Comment:
`typeByKClass` is doing `model.queryTypes[kClass]` with `queryTypes = queryTypeProxies + enums + scalars`. IMHO this is still fine as according to the spec the `__typename` is only available for unions, objects and interfaces - so no enums or scalars anyway.